src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -188,7 +188,10 @@ public void addMetaDataLine(final VCFHeaderLine headerLine) {
      * @return all of the VCF header lines of the ##contig form in order, or an empty list if none were present
      */
     public List<VCFContigHeaderLine> getContigLines() {
-        return Collections.unmodifiableList(contigMetaData);
+        List<VCFContigHeaderLine> contigLines = new ArrayList<>(contigMetaData.size());
+        // this must preserve input order
+        contigLines.addAll(contigMetaData.values());
+        return Collections.unmodifiableList(contigLines);
```

Comment:
Changing the return type has no effect on the underlying storage, so in my mind making the declared type more general is perfectly OK. The javadoc for the method is where I'd put the information about the method result being ordered, since that is a caller concern (whereas the underlying type, `List`, is not). But either way would work.